src/sim/probe/probe.cc

```diff
@@ -61,61 +61,39 @@ ProbeListenerObject::ProbeListenerObject(
 {
 }
 
-ProbeListenerObject::~ProbeListenerObject()
-{
-    for (auto l = listeners.begin(); l != listeners.end(); ++l) {
-        delete (*l);
-    }
-    listeners.clear();
-}
-
-ProbeListener::ProbeListener(ProbeManager *_manager, const std::string &_name)
-    : manager(_manager), name(_name)
-{
-    manager->addListener(name, *this);
-}
-
 ProbeListener::~ProbeListener()
 {
-    manager->removeListener(name, *this);
 }
 
 bool
-ProbeManager::addListener(std::string point_name, ProbeListener &listener)
+ProbeManager::addListener(std::string_view point_name, ProbeListener &listener)
 {
     DPRINTFR(ProbeVerbose, "Probes: Call to addListener to \"%s\" on %s.\n",
         point_name, object->name());
-    bool added = false;
-    for (auto p = points.begin(); p != points.end(); ++p) {
-        if ((*p)->getName() == point_name) {
-            (*p)->addListener(&listener);
-            added = true;
-        }
-    }
-    if (!added) {
+    if (ProbePoint *p = getProbePoint(point_name)) {
+        p->addListener(&listener);
+        return true;
+    } else {
```

Comment:
It has to be noted that in the previous implementation the loop was iterating over all probe points and the listener was being registered every time there was a match. In the new implementation you only register the listener once and return afterwards.
This is ok as long as probe names are unique. The question is, can you have multiple probes with the same name?

Reply:
Oh good catch! I didn't even consider more than one. Will restore the old logic in case anyone relies on this.